net/libfab/libfab.c

```diff
@@ -2790,6 +2708,27 @@ static int libfab_domain_params_get(struct m0_fab__ndom *fab_ndom)
 	return M0_RC(0);
 }
 
+static int libfab_buf_dom_dereg(struct m0_fab__buf *fbp)
+{
+	struct m0_fab__ndom *nd = fbp->fb_nb->nb_dom->nd_xprt_private;
+	int                  i;
+	int                  ret = 0;
+
+	for (i = 0; i < nd->fnd_seg_nr; i++) {
+		if (fbp->fb_mr.bm_mr[i] != NULL) {
+			ret = fi_close(&fbp->fb_mr.bm_mr[i]->fid);
```

Comment:
From fi_close() manpage:

Once unregistered, further access to the registered memory is not guaranteed. Active or queued operations that reference a memory region being closed may fail or result in accesses to invalid memory. Applications are responsible for ensuring that a MR is no longer needed prior to closing it.

So it seems it is not safe to call dereg() in buf_del(), while RDMA could still go on.